File: leetcode/169-Majority-Element/MajorityElement_001.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def majorityElement(self, nums):
        candidate1, count1 = None, 0
        for num in nums:
            if candidate1 == num:
                count1 += 1
            elif count1 == 0:
                count1 += 1
                candidate1 = num
            else:
                count1 -= 1
        return candidate1

File: leetcode/169-Majority-Element/java/MajorityElement_001.java
import java.util.Hashtable;

public class Solution {
    public int majorityElement(int[] num) {
        int len = num.length;
        Hashtable<Integer, Integer> tb = new Hashtable<Integer, Integer>();
        int max = 0;

        for (int i = 0; i < len; i++) {
            int digit = num[i];

            if (!tb.containsKey(digit)) {
                tb.put(digit, 1);
            } else {

                int val = tb.get(digit) + 1;
                tb.put(digit, val);
            }
        }

        int target = 0;

        for (Integer key : tb.keySet()) {
            int tmp = tb.get(key);
            if (tmp > max) {
                max = tmp;
                target = key;
            }
        }

        return target;

    }
}
